src/k2/cmd/tpcc/transactions.h

```diff
@@ -1175,3 +1175,157 @@ class DeliveryT : public TPCCTxn
     std::vector<int32_t> _O_C_ID = std::vector<int32_t>(10, -1); // ORDER table info: customer ID
     std::vector<std::decimal::decimal64> _OL_SUM_AMOUNT = std::vector<std::decimal::decimal64>(10, -1); // ORDER-LINE table info: sum of all amount
 };
+
+class StockLevelT : public TPCCTxn
+{
+public:
+    StockLevelT(RandomContext& random, K23SIClient& client, int16_t w_id, int16_t d_id) :
+                        _random(random), _client(client), _w_id(w_id), _d_id(d_id) {
+        _threshold = _random.UniformRandom(10, 20);
+        _low_stock = 0;
+
+        _failed = false;
+    }
+
+    future<bool> run() override {
+        K2TxnOptions options{};
+        options.deadline = Deadline(5s);
+        return _client.beginTxn(options)
+        .then([this] (K2TxnHandle&& txn) {
+            _txn = std::move(txn);
+            return runWithTxn();
+        }).handle_exception([] (auto exc) {
+            K2LOG_W_EXC(log::tpcc, exc, "Failed to start txn");
+            return make_ready_future<bool>(false);
+        });
+    }
+
+private:
+    future<bool> runWithTxn() {
+        return getNextOrder()
+        .then([this] (int64_t d_next_order_id) {
+            K2LOG_D(log::tpcc, "StockLevel txn got next order no = {}", d_next_order_id);
+            return getItemIDs(d_next_order_id);
+        })
+        .then([this] (std::set<int32_t> && item_ids){
+            K2LOG_D(log::tpcc, "StockLevel txn got item_ids - no of items = {}", item_ids.size());
+            return getStockQuantity(std::move(item_ids)).discard_result();
```

Comment:
Same as below, getStockQuantity returns a void future so no need for discard_result.